linetools/isgm/emsystem.py

```diff
@@ -0,0 +1,536 @@
+        elif isinstance(inp, AbsLine):
+            return self.get_comp_from_absline(inp)
+        else:
+            raise IOError("Bad input to get_component method")
+
+    def get_comp_from_absline(self, aline):
+        """ Returns the component that holds the input AbsLine
+
+        Parameters
+        ----------
+        aline : AbsLine
+
+        Returns
+        -------
+        comp -- AbsComponent object that holds this AbsLine
+        """
+        # Loop on components
+        for comp in self._components:
+            # Is the line present?
+            try:
+                idx = comp._abslines.index(aline)
+            except ValueError:
+                pass
+            else:
+                return comp
+        # Raise error?
+        warnings.warn("Input absorption line is not in any component")
+        return None
+
+    def measure_restew(self, spec=None, **kwargs):
+        """ Measure rest-frame EWs for lines in the AbsSystem
+        Parameters
+        ----------
+        spec : XSpectrum1D, optional
+        kwargs
+
+        Returns
+        -------
+
+        """
+        # Grab Lines
+        abs_lines = self.list_of_abslines()
+        # Loop
+        for iline in abs_lines:
+            # Fill in spec?
+            if spec is not None:
+                iline.analy['spec'] = spec
+            # Measure
+            iline.measure_restew(**kwargs)
+
+    def measure_aodm(self, spec=None, **kwargs):
+        """ Measure ADOM columns for the list of lines
+        Note: Components are *not* updated by default
+
+        Parameters
+        ----------
+        spec : XSpectrum1D, optional
+        kwargs
+
+        Returns
+        -------
+
+        """
+        # Grab Lines
+        abs_lines = self.list_of_abslines()
+        # Loop
+        for iline in abs_lines:
+            # Fill in spec?
+            if spec is not None:
+                iline.analy['spec'] = spec
+            # Measure
+            iline.measure_aodm(**kwargs)
+        #
+        print("You may now wish to update the component column densities with update_component_colm()")
+
+    def update_component_colm(self, **kwargs):
+        """ Synthesize/update column density measurements for components
+
+        Parameters
+        ----------
+        kwargs
+
+        Returns
+        -------
+
+        """
+        for comp in self._components:
+            comp.synthesize_colm(**kwargs)
+
+    def stack_plot(self, pvlim=None, **kwargs):
+        """Show a stack plot of the system, if spec are loaded
+        Assumes the data are normalized.
+
+        Parameters
+        ----------
+        pvlim : Quantities, optional
+          Over-ride system vlim for plotting
+        """
+        from linetools.analysis import plots as ltap
+        if pvlim is not None:
+            vlim = pvlim
+        else:
+            vlim = self.vlim
+        ltap.stack_plot(self.list_of_abslines(), vlim=vlim, **kwargs)
+
+    def to_dict(self):
```

Comment:
Does this work?

Reply:
Just checked that this works! There is also a test for it now.